numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
`Py_None` is garantueed to exist and as long as it doesn't escape to the user and it isn't decref'd everything should be fine. `PyObject_CallFunction` increments it's reference count for the exception so what the user gets is a new reference.

Using `NULL` seems cleaner though.